components/fxa-client/src/error.rs

```diff
@@ -193,7 +197,28 @@ impl GetErrorHandling for Error {
             Error::RequestError(_) => {
                 ErrorHandling::convert(crate::FxaError::Network).log_warning()
             }
+            // Callback errors should be handled by the calling code, for example
+            // `CallbackError::Cancelled`.  Report any that reach this stage
+            Error::CallbackError(_) => {
+                ErrorHandling::convert(crate::FxaError::Other).report_error("fxa-callback-error")
+            }
             _ => ErrorHandling::convert(crate::FxaError::Other).log_warning(),
         }
     }
 }
+
+/// Errors thrown by by callback interfaces
+#[derive(Debug, thiserror::Error)]
+pub enum CallbackError {
+    #[error("Cancelled")]
+    Cancelled,
+
+    #[error("CallbackError: {reason}")]
+    Other { reason: String },
+}
```

Comment:
Seems fine to me to have an error enum specifically for things we might expect callbacks to do - what's your concern?

To my mind, calling `Cancelled` an error is a bit of a smell - it's almost like we want an enum much like `Result<>` but with a 3rd option for cancelled to reinforce it's an expected non-error state - but I don't think we should actually do that :)

Reply:
James actually mentioned the same thing.  Maybe a `Result<Option<T>>` is better.

That gets a bit weird, since then `CallbackError` only has one variant.  I guess that's just more motivation for the error work that you're doing.

Reply:
I think the biggest issue with `Result<Option<T>>` is that it means that consumers have to return `null`  to signal cancelled, which doesn't seem right.  Also, they have several nested functions, than it might be more idiomatic to let a `Cancelled` error bubble up than for each of them to handle `null`.

Reply:
I don't like `Result<Option<T>` for the same reason. `enum CallbackResult ...` is what I had in mind, but I suspect the cure will be worse than the disease...